internal/provider/virtual_cluster.go

```diff
@@ -5,18 +5,17 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-type apiKeyModel struct {
-	Name      types.String `tfsdk:"name"`
-	Key       types.String `tfsdk:"key"`
-	CreatedAt types.String `tfsdk:"created_at"`
-}
+const (
+	virtualClusterTypeBYOC       = "byoc"
+	virtualClusterTypeServerless = "serverless"
+)
 
 // virtualClusterModel maps virtual cluster schema data.
 type virtualClusterDataSourceModel struct {
 	ID            types.String `tfsdk:"id"`
 	Name          types.String `tfsdk:"name"`
 	Type          types.String `tfsdk:"type"`
-	AgentKeys     *apiKeyModel `tfsdk:"agent_keys"` // Null in serverless VCs.
+	AgentKeys     *apiKeyModel `tfsdk:"agent_keys"` // Hack: pointer, not slice, so it can be null for serverless VCs.
```

Comment:
I don't understand this comment. Why can't you return a nil slice for serverless?

Reply:
I was too sleepy to

Reply:
Why not a slice directly? `nil` is a valid slice value

Reply:
Yes but not for Terraform! It was complaining about seeing null values.

Reply:
ok!